components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementEndpointUtil.java

```diff
@@ -506,8 +506,9 @@ public static String getURLEncodedCallback(String callbackUrl) throws URISyntaxE
      */
     public static String buildEndpointUrl(String path) {
 
-        String endpointUrl = IdentityManagementServiceUtil.getInstance().getServiceContextURL()
-                .replace(IdentityManagementEndpointConstants.UserInfoRecovery.SERVICE_CONTEXT_URL, "");
+        String serviceContextURL = IdentityManagementServiceUtil.getInstance().getServiceContextURL();
+        String endpointUrl = replaceLastOccurrence(serviceContextURL,
```

Comment:
We have fixed a similar [issue](https://github.com/wso2/product-is/issues/4251)[1] with the PR https://github.com/wso2/carbon-identity-framework/pull/2031. There, we have introduced a new util method to construct the URL replacing the  "/services".[2] You can use the same util method here also.
 
[1] https://github.com/wso2/product-is/issues/4251
[2] https://github.com/wso2/carbon-identity-framework/pull/2031/files#diff-b4accc0e6ab6ae61eafa680e285fe198R458

Reply:
In this PR an improvement is introduced to [1]
The initial fix replaces `/services` string in the url. If the hostname starts with services, i.e `services.foo.bar,` then the first services get replaced. 

With this fix, I find the last occurrence of the /services and replace the string.

[1] https://github.com/wso2/carbon-identity-framework/pull/2031/files#diff-b4accc0e6ab6ae61eafa680e285fe198R458